packages/forms/src/UIForm/fields/NestedListView/NestedListView.utils.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Build the ListView component props to match current state's values
+ * @param {Array} items
+ * @param {Object} value
+ * @param {String} searchCriteria
+ * @param {Array} toggledChildren
+ * @returns {Object}
+ */
+export function getItemsProps(items, value, searchCriteria, toggledChildren) {
+	const displayedItems = items.reduce((finalItems, item) => {
+		const { children } = item;
+		const itemValue = value[item.key] || [];
+
+		// Filter children items if search criteria has been provided
+		const finalChildren = searchCriteria
+			? children.filter(({ label }) => label.toLowerCase().includes(searchCriteria.toLowerCase()))
+			: children;
+
+		if (finalChildren.length > 0) {
+			const checked = item.key in value && children.some(child => itemValue.includes(child.value));
+
+			finalItems.push({
+				...item,
+				checked,
+				expanded: toggledChildren.includes(item.toggleId),
+				children: finalChildren.map(child => ({
+					...child,
+					checked: itemValue.includes(child.value),
+				})),
+			});
+		}
+
+		return finalItems;
+	}, []);
+
+	return {
+		displayedItems,
+		items,
+		searchCriteria,
+		toggledChildren,
```

Comment:
the last 3 object properties are not calculated in the function. They shouldn't be returned. In fact this function seems to only calculate the `displayedItems` so should return only this value and have a more meaningful name according to that unique calculated value.

Reply:
ps: The other values can be given to the state or the instance attributes directly without hiding the changes.